src/de/key.rs

```diff
@@ -0,0 +1,458 @@
+use crate::de::str2bool;
+use crate::encoding::Decoder;
+use crate::errors::serialize::DeError;
+use crate::name::QName;
+use serde::de::{DeserializeSeed, Deserializer, EnumAccess, VariantAccess, Visitor};
+use serde::{forward_to_deserialize_any, serde_if_integer128};
+use std::borrow::Cow;
+
+macro_rules! deserialize_num {
+    ($method:ident, $visit:ident) => {
+        fn $method<V>(self, visitor: V) -> Result<V::Value, Self::Error>
+        where
+            V: Visitor<'de>,
+        {
+            visitor.$visit(self.name.parse()?)
+        }
+    };
+}
+
+/// Decodes raw bytes using the deserializer encoding.
+/// The method will borrow if encoding is UTF-8 compatible and `name` contains
+/// only UTF-8 compatible characters (usually only ASCII characters).
+#[inline]
+fn decode_name<'n>(name: QName<'n>, decoder: Decoder) -> Result<Cow<'n, str>, DeError> {
+    let local = name.local_name();
+    Ok(decoder.decode(local.into_inner())?)
+}
+
+/// A deserializer for xml names of elements and attributes.
+///
+/// Used for deserialize values from:
+/// - attribute names (`<... name="..." ...>`)
+/// - element names (`<name>...</name>`)
+///
+/// Converts name to the identifier string by following rules:
+///
+/// - if that is an [`attribute`] name, put `@` to the identifier
+/// - put the decoded [`local_name()`] of a name to the identifier
```

Comment:
I'm not actually sure what you're trying to say here

Reply:
I try say here that XML name stored in struct's `name` field would be converted to the "identifier" (in terms of serde's `Deserializer::deserialize_identifier`) of the Rust field name using the described rules. So Rust (actually, serde) name of field for an attribute `name` should be `@name` and for element `<name>` should be `name`.